src/Containers/NoteForm/NoteForm.js

```diff
@@ -1,43 +1,74 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { setError } from '../../actions';
+import API from '../../utils/api';
 
 class NoteForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      ...this.props.note
+      id: this.props.id,
+      title: this.props.title,
+      issues: this.props.issues
     }
   }
 
-  handleChange = (e) => {
-    const index = this.state.issues.findIndex((issue) => issue.id == e.target.id)
-    console.log('that', index)
-    const issues = this.state.issues.slice()
-    issues[index].body = e.target.value
-    this.setState({ issues })
+  handleBodyChange = (e) => {
+    const index = this.state.issues.findIndex((issue) => issue.id == e.target.parentElement.parentElement.id);
+    const issues = this.state.issues.slice();
+    issues[index].body = e.target.value;
+    this.setState({ issues });
   } 
 
+  handleTitleChange = (e) => {
+    this.setState({ title: e.target.value });
+  } 
+
+  toggleIssueCompletion = (e) => {
+    const index = this.state.issues.findIndex((issue) => issue.id == e.target.parentElement.parentElement.id);
+    const issues = this.state.issues.slice();
+    issues[index].completed = !issues[index].completed;
+    this.setState({ issues });
+  } 
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { setError } = this.props;
+    try {
+      this.formRef.reset();
+    } catch (error) {
+      setError(error)
+    }
+  }
+
+  showIssues = (completed) => {
+    const { issues } = this.state;
+    return issues.filter(issue => issue.completed === completed).map(issue =>
+      <li key={issue.id} id={issue.id}>
+        <span>
+          <i onClick={this.toggleIssueCompletion} className="fas fa-square"></i>
+          <input onChange={this.handleBodyChange} value={issue.body}></input>
+        </span>
+        <i className="fas fa-times"></i>
+      </li>
+    )
+  }
+
   render() {
-    const { title, issues } = this.state
+    const { title } = this.state;
+    const incompleteIssues = this.showIssues(false);
+    const completeIssues = this.showIssues(true);
+
     return (
       <div className='overlay-div'>
-        <form className='note-pop-up'>
-          <h3>{title}</h3>
-          <ul>
-            {
-              issues.filter(issue => !issue.completed).map(issue => <li><span><i class="fas fa-square"></i> <input id={issue.id} onChange={this.handleChange} value={issue.body}></input></span> <i class="fas fa-times"></i></li>)
-            }
-          </ul>
+        <form className='note-pop-up' onSubmit={this.handleSubmit} ref={(el) => this.formRef = el}>
+          <input onChange={this.handleTitleChange} value={title}></input>
```

Comment:
Does ref={(el) => this.formRef = el} just clear the input fields?

Reply:
Yeah, I'm not sure if it's necessary but I've used it previously on forms so I put it in there. We can try it without the ref/form reset too.